omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,300 @@
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.RefRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+
+@Resource(name = RestConstants.VERSION_1 + IdgenRestController.IDGEN_NAMESPACE + "/identifiersource", supportedClass = IdentifierSource.class, supportedOpenmrsVersions = {
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*" , "2.1.*" })
+public class IdentifierSourceResource extends DelegatingCrudResource<IdentifierSource>{
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		
+		DelegatingResourceDescription description = null;
+		
+		if (rep instanceof RefRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("display");
+			description.addSelfLink();
+			return description;
+		}
+		
+		if (rep instanceof DefaultRepresentation || rep instanceof FullRepresentation) {
+			description = new DelegatingResourceDescription();
+			description.addProperty("uuid");
+			description.addProperty("name");
+			description.addProperty("description");
+			description.addProperty("identifierType");
+			
+			// links
+			description.addSelfLink();
+			if (rep instanceof DefaultRepresentation) {
+				description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			}
+		}
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("name");
+		description.addProperty("description");
+		description.addProperty("identifierType");
+		
+		return description;
+	}
+	
+	@PropertyGetter("identifierType")
+	public String getidentifierType(IdentifierSource identifierSource) {
```

Comment:
What happens if you do not have this?

Reply:
i get the following errors: `ERROR!
org.codehaus.jackson.map.JsonMappingException: Conflicting getter definitions for property "retired": org.openmrs.PatientIdentifierType_$$_javassist_23#getRetired(0 params) vs org.openmrs.PatientIdentifierType_$$_javassist_23#isRetired(0 params)`

Reply:
When do you get that error? In other words, what are the steps to reproduce it?

Reply:
If I remove the property getter and send a get request the error occurs.

Reply:
Which request are you sending? Can i look at it?

Reply:
I have recreated the error by commenting out the PropertyGetter. 

In IdentifierSourceRestControllerTest.java: The methods shouldGetAnIdentifierSourceByUuid(), shouldGetDefaultByUuid() and shouldGetDefaultByUuid() fail with the above error.

See the Travis log.

Reply:
Oh i see! Go ahead and put it back.

Reply:
fixed